hail/python/hailtop/batch/backend.py

```diff
@@ -309,18 +309,51 @@ class ServiceBackend(Backend):
     >>> b.run() # doctest: +SKIP
     >>> service_backend.close()
 
+    Instead of a bucket, a full path may be specified for the remote temporary directory:
+
+    >>> service_backend = ServiceBackend('my-billing-account',
+    ...                                  remote_tmpdir='gs://my-bucket/temporary-files/')
+    >>> b = Batch(backend=service_backend)
+    >>> b.run() # doctest: +SKIP
+    >>> service_backend.close()
+
     Parameters
     ----------
     billing_project:
         Name of billing project to use.
     bucket:
-        Name of bucket to use.  Should not include the ``gs://``
-        prefix.
+        Name of bucket to use. Should not include the ``gs://`` prefix. Cannot be used with
+        remote_tmpdir. Temporary data will be stored in the "/batch" folder of this
+        bucket. Using this parameter as a positional argument is deprecated.
+    remote_tmpdir:
+        Temporary data will be stored in this google cloud storage folder. Cannot be used with bucket. Example: "gs://
+
     """
 
     def __init__(self,
-                 billing_project: str = None,
-                 bucket: str = None):
+                 *args,
+                 bucket: Optional[str] = None,
+                 remote_tmpdir: Optional[str] = None
+                 ):
+        billing_project: Optional[str]
+        if len(args) > 2:
+            raise TypeError(f'ServiceBackend() takes 2 positional arguments but {len(args)} were given')
+        if len(args) == 2:
+            billing_project = args[0]
+            if bucket is not None:
+                raise TypeError('ServiceBackend() got multiple values for argument \'bucket\'')
+            bucket = args[1]
+            warnings.warn('Use of deprecated positional argument \'bucket\' in ServiceBackend(). Specify \'bucket\' as a keyword argument instead.')
+        elif len(args) == 1:
+            billing_project = args[0]
+        elif len(args) == 0:
+            billing_project = None
+
+        if remote_tmpdir is not None and bucket is not None:
+            raise ValueError('Cannot specify both remote_tmpdir and bucket in ServiceBackend()')
+        if remote_tmpdir is None and bucket is None:
```

Comment:
What happened to trying to get the bucket from the user config here? I think this logic is further down below

Reply:
fixed